src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php

```diff
@@ -121,28 +113,10 @@ public function finishView(FormView $view, FormInterface $form, array $options)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        // BC clause for the "intention" option
-        $csrfTokenId = function (Options $options) {
-            return $options['intention'];
-        };
-
-        // BC clause for the "csrf_provider" option
-        $csrfTokenManager = function (Options $options) {
-            if ($options['csrf_provider'] instanceof CsrfTokenManagerInterface) {
-                return $options['csrf_provider'];
-            }
-
-            return $options['csrf_provider'] instanceof CsrfTokenManagerAdapter
-                ? $options['csrf_provider']->getTokenManager(false)
-                : new CsrfProviderAdapter($options['csrf_provider']);
-        };
-
         $resolver->setDefaults(array(
             'csrf_protection' => $this->defaultEnabled,
             'csrf_field_name' => $this->defaultFieldName,
             'csrf_message' => 'The CSRF token is invalid. Please try to resubmit the form.',
-            'csrf_token_manager' => $csrfTokenManager,
```

Comment:
This and the next option should not be removed. Instead you should remove `csrf_provider` and `intention`.